modules/core/src/main/scala/gem/Step.scala

```diff
@@ -9,6 +9,37 @@ sealed abstract class Step[A] extends Product with Serializable {
   def instrument: A
 }
 
+object Step {
+  final case class Location(loc: List[Int]) {
+    override def toString: String =
+      loc.mkString("{", ",", "}")
+  }
+
+  object Location {
+    def apply(is: Int*): Location =
+      Location(is.toList)
+
+    private val LocationString = """^\{([0-9]+(?:,[0-9]+)*)\}$""".r
+
+    def parse(s: String): Option[Location] =
+      s match {
+        case LocationString(ds) => Some(Location(ds.split(',').toList.map(_.toInt)))
+        case _                  => None
+      }
+
+    def unsafeParse(s: String): Location =
```

Comment:
There is a plain `parse` method for that.  In the end I don't think I really need to parse from `String` anyway so I'd probably just delete these methods altogether.